hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java

```diff
@@ -3593,11 +3593,13 @@ public void writeXml(Writer out) throws IOException {
    * </ul>
    * @param propertyName xml property name.
    * @param out the writer to write to.
+   * @param config configuration.
    * @throws IOException raised on errors performing I/O.
    */
-  public void writeXml(@Nullable String propertyName, Writer out)
+  public void writeXml(@Nullable String propertyName, Writer out, Configuration config)
```

Comment:
We keep calling this with null, keep a version with the old signature that passes null.

Reply:
I will refactor part of the code.